libraries/AP_ExternalAHRS/AP_ExternalAHRS_InertialLabs.cpp

```diff
@@ -0,0 +1,659 @@
+            state2.gnss_info_short = u.gnss_info_short;
+            break;
+        }
+        case MessageType::GNSS_NEW_DATA: {
+            CHECK_SIZE(u.gnss_new_data);
+            state2.gnss_new_data = u.gnss_new_data;
+            break;
+        }
+        case MessageType::GNSS_JAM_STATUS: {
+            CHECK_SIZE(u.gnss_jam_status);
+            state2.gnss_jam_status = u.gnss_jam_status;
+            break;
+        }
+        case MessageType::DIFFERENTIAL_PRESSURE: {
+            CHECK_SIZE(u.differential_pressure);
+            airspeed_data.differential_pressure = u.differential_pressure*1.0e-4;
+            break;
+        }
+        case MessageType::TRUE_AIRSPEED: {
+            CHECK_SIZE(u.true_airspeed);
+            state2.true_airspeed = u.true_airspeed*0.01;
+            break;
+        }
+        case MessageType::WIND_SPEED: {
+            CHECK_SIZE(u.wind_speed);
+            state2.wind_speed = u.wind_speed.tofloat().rfu_to_frd()*0.01;
+            break;
+        }
+        case MessageType::AIR_DATA_STATUS: {
+            CHECK_SIZE(u.air_data_status);
+            state2.air_data_status = u.air_data_status;
+            break;
+        }
+        case MessageType::SUPPLY_VOLTAGE: {
+            CHECK_SIZE(u.supply_voltage);
+            state2.supply_voltage = u.supply_voltage*0.01;
+            break;
+        }
+        case MessageType::TEMPERATURE: {
+            CHECK_SIZE(u.temperature);
+            // assume same temperature for baro and airspeed
+            baro_data.temperature = u.temperature*0.1;
+            airspeed_data.temperature = u.temperature*0.1;
+            break;
+        }
+        case MessageType::UNIT_STATUS2: {
+            CHECK_SIZE(u.unit_status2);
+            state2.unit_status2 = u.unit_status2;
+            break;
+        }
+        }
+
+        if (msg_len == 0) {
+            // got an unknown message
+            GCS_SEND_TEXT(MAV_SEVERITY_INFO, "InertialLabs: unknown msg 0x%02x", unsigned(mtype));
+            buffer_ofs = 0;
+            return false;
+        }
+        message_ofs += msg_len;
+
+        if (msg_len == 0 || need_re_sync) {
+            re_sync();
+            return false;
+        }
+    }
+
+    if (h->msg_len != message_ofs-buffer) {
+        // we had stray bytes at the end of the message
+        re_sync();
+        return false;
+    }
+
+    if (GOT_MSG(ACCEL_DATA_HR) &&
+        GOT_MSG(GYRO_DATA_HR)) {
+        AP::ins().handle_external(ins_data);
+    }
+    if (GOT_MSG(GPS_INS_TIME_MS) &&
+        GOT_MSG(NUM_SATS) &&
+        GOT_MSG(GNSS_POSITION) &&
+        GOT_MSG(GNSS_NEW_DATA) &&
+        GOT_MSG(GNSS_EXTENDED_INFO) &&
+        state2.gnss_new_data != 0) {
+        uint8_t instance;
+        if (AP::gps().get_first_external_instance(instance)) {
+            AP::gps().handle_external(gps_data, instance);
+        }
+        if (gps_data.satellites_in_view > 3) {
+            if (last_gps_ms == 0) {
```

Comment:
Will this check won't work if the GPS is intermittently getting lock? Perhaps a ` uint8_t last_gps_satellites` comparison would be more appropriate.

Reply:
fixed it to use the status bits